dimod/sampleset.py

```diff
@@ -450,6 +459,58 @@ def infer_vartype(samples_like):
     raise ValueError("given samples_like is of an unknown vartype")
 
 
+def drop_variables(sampleset: 'SampleSet', variables: Iterable[Variable]) -> 'SampleSet':
+    """Return a new sample set with the given variables removed.
+
+    Args:
+        sampleset: A sample set.
+        variables: The variables to be dropped. Can contain variables not in
+            the sample set.
+
+    Returns:
+        A new sampleset without the given variables. The energies, info
+        and other data vectors will be the same as in the given sample set.
+
+    """
+    return keep_variables(sampleset, sampleset.variables - variables)
+
+
+def keep_variables(sampleset: 'SampleSet', variables: Iterable[Variable]) -> 'SampleSet':
+    """Return a new sample set with only the given variables.
+
+    Args:
+        sampleset: A sample set.
+        variables: The variables to be kept. Must be a subset of the variables
+            in the sample set.
```

Comment:
For symmetry with `drop_variables`, why not allow variables not in the sampleset?

Reply:
In this case the asymmetry is deliberate. I think of it as "drop any of these variables" and "keep all of these variables". So if I was able to provide variables not in the sample set to keep, I would expect the created sample set to have all of the ones specified.

Reply:
Sure, I get that. Personally, I would appreciate consistency, so at least for drop to require a subset... But I really don't have strong feeling about it. :shrug: